aten/src/ATen/native/cuda/DistanceKernel.cu

```diff
@@ -335,12 +335,19 @@ void cdist_backward_kernel_impl(Tensor& result, const Tensor& grad, const Tensor
   const int block_x = 64;
   const int block_y = 16;
   const int grid_x = (m + block_x * 8 - 1) / (block_x * 8);
-  const int grid_y = (dist.numel() + block_y - 1) / block_y;
 
-  const dim3 grid(grid_x, grid_y);
+  const int64_t count = dist.numel();
```

Comment:
if count is 0 this would also lead to invalid launch configuration - is it checked anywhere before? If count=0 you should do an early return, without launching a kernel

Reply:
Emmm I think this behaviour is already avoided by conditional checking in line 326. The situation with count being 0 should be identical to that the output (Tensor dist) has no elements neither, which can be leaded by:
1. tensor x1 has no elements (x1.numel() == 0)
2. tensor x2 has no elements (x2.numel() == 0)
3. both of 1 and 2

Anyway, adding this checking gives double insurances over this, and I can modify it if you like.

Besides I cleaned the commited lines and keep the std::sqrt approach. The former is unsufficient with some extreme cases (e.g. 65537).

Reply:
Ah... seems that you approach is also sufficient.

Reply:
Cool, if there's an existing check for numel=0, no need to add it here.